### PR TITLE
dtj chapter 9

### PR DESCRIPTION
**Pull request by thatguy59**

### CODE REVIEW

File: chapter09/dtj/ex3 [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+use warnings;
+use strict;
+
+if ( !$ARGV[0] || !open INPUT, "<", $ARGV[0] ) {
+  die "Could not open file for reading\n";
+}
+
+open OUTPUT, ">", $ARGV[0] . ".out";
+
+# Use something random, as it's less likely to appear in the input
```
Comment (MarkWignall) [inline chapter09/dtj/ex3:13]: What?

Comment (thatguy59) [inline chapter09/dtj/ex3:13]: https://xkcd.com/221/

I wanted something that was less likely to be in the text already as the temporary thing.

File: chapter09/dtj/ex5 [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+use strict;
+use warnings;
+
+$^I = ".bak";
+
+my $copyright = "## Copyright (C) 2015 by dtj";
+
+while (<>) {
+	if (!/\#\# Copyright \(C\)/) {
```
Comment (codavex) [inline chapter09/dtj/ex5:12]: Nice, much simpler than my version

Comment (qianliok) [inline chapter09/dtj/ex5:12]: Yeah, I like this approach every much.

File: chapter09/dtj/ex2 [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+use strict;
+
+if ( !$ARGV[0] || !open INPUT, "<", $ARGV[0] ) {
+  die "Could not open file for reading\n";
+}
+
+open OUTPUT, ">", $ARGV[0] . ".out";
+
+while (<INPUT>) {
+    s/fred/larry/gi;
```
Comment (qianliok) [inline chapter09/dtj/ex2:14]: larry should be Larry, I made the same mistake :smile:

File: chapter09/dtj/ex4 [hunk 4]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/perl
+## Copyright (C) 2015 by dtj
+
+use strict;
+use warnings;
+
+$^I = ".bak";
+
+while (<>) {s
```
Comment (qianliok) [inline chapter09/dtj/ex4:9]: s?

**Pull request merged**